integrations/terraform/examples/resources/teleport_discovery_config/resource.tf

```diff
@@ -0,0 +1,74 @@
+# Teleport Discovery Config
+#
+# Discovery Config resources define matchers for the Teleport Discovery Service.
+# The Discovery Service automatically discovers and enrolls cloud resources
+# (EC2 instances, RDS databases, EKS clusters, Azure VMs, etc.) into your
+# Teleport cluster.
+#
+# Each Discovery Config is associated with a discovery_group. Discovery Services
+# load matchers from Discovery Configs that share the same discovery_group.
+
+# Example: AWS Discovery Config for EC2 instances and RDS databases
+resource "teleport_discovery_config" "aws_example" {
+  header = {
+    metadata = {
+      name        = "aws-discovery"
+      description = "Discover AWS EC2 instances and RDS databases"
+      labels = {
+        env = "production"
+      }
+    }
+    version = "v1"
+  }
+
+  spec = {
+    discovery_group = "aws-prod"
+
+    aws = [{
+      types   = ["ec2", "rds"]
+      regions = ["us-west-2", "us-east-1"]
+      tags = {
+        "env" = ["prod", "production"]
+      }
+      install_params = {
+        join_method = "iam"
+        join_token  = "aws-discovery-token"
+        script_name = "default-installer"
+      }
```

Comment:
There are a couple of values that are set when the `install_params` is not set in the `teleport.yaml/discovery_service.aws[].install` or in `discovery_config.spec.aws[].install`.


As an example, if the install_params is not set, it will set the:
- enroll_mode to `script` enum value
- install_teleport to `true`

I'm not sure we can do the same logic here, but if possible let's at least ensure the example value is set to those values.